### ISSUE TITLE
Nebula fails if osquery is not able to run a query

### ISSUE THREAD

**Issue by yassingh** [hubblestack/hubble#210]
In Ubuntu 14, osquery fails when it tries to execute the following query: 
```
- query_name: deb_packages
  query: SELECT t.unix_time AS query_time, deb.name, deb.version, deb.revision, deb.source AS package_source, deb.size, deb.arch FROM deb_packages AS deb JOIN time AS t;
```

I get the following error if I try to run this query manually using osqueryi:

```
osquery: error: failed to open package info file ‘/var/lib/dpkg//status’ for reading: Permission denied
terminate called without an active exception
Segmentation fault
```

The result object returned to nebula module does not contain any 'data' key, so the code fails and logs are not pushed for the whole group of nebula queries. Logs:

```
2017-09-26 20:15:08 [hubblestack.daemon][ERROR   ] Error executing schedule
Traceback (most recent call last):
  File "hubblestack/daemon.py", line 94, in main
  File "hubblestack/daemon.py", line 209, in schedule
  File "/opt/hubble/hubble-libs/hubblestack/extmods/modules/nebula_osquery.py", line 202, in queries
    for result in query_ret['data']:
KeyError: 'data'
```

**Comment by basepi** [comment]
Good catch! We should fix nebula to not error out all the way if there's a bad query.

(Also I edited your issue -- you can use triple backticks to do code blocks, rather than backticking each line manually. Click edit on your issue to see how it looks.)

**Comment by yassingh** [comment]
Thanks @basepi 👍  . The triple backticks trick is really what I was looking for while logging the bug.

**Comment by fossam** [comment]
@yassingh Are there any special circumstances that might go into replicating this issue?
In preliminary testing I get this:
```
$ cat /etc/*release | head -2
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
$ /opt/osquery/osqueryi --version
osqueryi version 2.8.0
$ /opt/osquery/osqueryi "SELECT t.unix_time AS query_time, deb.name, deb.version, deb.revision, deb.source AS package_source, deb.size, deb.arch FROM deb_packages AS deb JOIN time AS t LIMIT 3;"
+------------+-----------+-------------------+----------+----------------+------+-------+
| query_time | name      | version           | revision | package_source | size | arch  |
+------------+-----------+-------------------+----------+----------------+------+-------+
| 1507142081 | adduser   | 3.113+nmu3ubuntu3 |          |                | 644  | all   |
| 1507142081 | apt       | 1.0.1ubuntu2.17   |          |                | 3578 | amd64 |
| 1507142081 | apt-utils | 1.0.1ubuntu2.17   |          | apt            | 668  | amd64 |
+------------+-----------+-------------------+----------+----------------+------+-------+
```

**Comment by basepi** [comment]
Even if we fix the original error in the query, nebula should not fail so hard, so I'll leave this open either way.

**Comment by yassingh** [comment]
@fossam 
I encountered this issue with osquery 2.6.0 and 2.7.0 . I didn't try osquery 2.8.0 ( i am not sure that it's a stable release ).
Besides that , I tried this query on two Ubuntu 14 AMIs available to me. Perhaps those AMIs are Adobe specific, so I didn't mention them.

**Comment by fossam** [comment]
@yassingh Can you test this in a default (no flavors) 14.04 ubuntu (vmware/virtualbox/etc VM) install and tell me if you can replicate it. I can't but I'd like to know if I am overlooking something.

osquery 2.8.0 is stable enough to withstand this test. changes between 2.8.0 and 2.8.1 and minimal and should not affect this issue.

**Comment by yassingh** [comment]
I didn't get time to test this issue on default hubble 14.04 ( actually i ran into some sort of error while installing hubble )

But I installed latest hubble (2.2.8.1 ) on the ubuntu machine on which i was getting osquery error earlier. **Now osquery is able to run this nebula query properly** .  This hubble (2.2.8.1) build comes with osquery 2.8 whereas the older build came with osquery 2.6.
I guess this issue has been resovled in osquery 2.8.

 However, if time permits, I will try to reproduce this issue on default hubble 14.04 as well. Will let you know about the results when I do so. 

( FYI, i am getting hubble installation error that "git must be installed" but I have already installed and updated git on the machine )
